### PR TITLE
Adding tests for org.openmrs.util.Format in org.openmrs.util.FormatTe…

### PR DESCRIPTION
**Pull request by winglam**
…st. The added tests calls org.openmrs.util.Format.format with nulls and valid arguments. The following tests fail because it is expecting org.openmrs.util.Format.format to return null. format_shouldAllowAllNullParameters, format_shouldAllowLocaleTypeNullParameters, format_shouldAllowLocaleNullParameters, and format_shouldAllowDateLocaleNullParameters gets a NullPointerException instead and format_shouldReturnDateWithNullType and format_shouldAllowDateNullParameters receives an empty string as the return.

A bug report related to this pull request can be found on: 
https://issues.openmrs.org/browse/TRUNK-4849

### CODE REVIEW

**Comment by dkayiwa** [comment]
@winglam did you take a look at this? https://wiki.openmrs.org/display/docs/Unit+Testing+With+at-should+Annotation

**Comment by winglam** [comment]
@dkayiwa Thanks for pointing out the @should annotation. I have added that, other documentation, null checks to org.openmrs.util.Format and changed the tests in org.openmrs.util.FormatTest to respect the documentation.

**Comment by dkayiwa** [comment]
Can you squash these commits into one? https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by winglam** [comment]
I have squashed the commits.

File: api/src/main/java/org/openmrs/util/Format.java [hunk 1]
```diff
@@ -62,8 +62,25 @@ public static String format(Date date) {
...
+	 * The string can contain only the date (month, day and year), only the time (hours, minutes, seconds) or as a
+	 * timestamp (both date and time).
+	 *
+	 * @param date input date to format as a string
+	 * @param locale input locale to determine how to format the date
+	 * @param type input type to determine how much information from the date is returned
+	 * @return empty string if one of the parameters is null. Otherwise a string object for the date such that it is
+	 * formatted according to locale and the amount of information it contains is determined by type.
+	 * @should not fail with null date
+	 * @should not fail with null locale
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/util/Format.java]: Did you add a should annotation for each and every test you added?

Comment (winglam) [inline api/src/main/java/org/openmrs/util/Format.java]: No I did not. However I believe these three annotations are sufficient for the 8 tests that I have added. 3 of the 8 tests test the case when one of these parameters are null. Another 3 of the 8 tests test the case when two of the parameters are null. The remaining two tests test the case when all of them are null or none of them are null. The should annotations I have added implies that regardless of whether any of the parameters are null or not, the method should not fail.

Comment (dkayiwa) [inline api/src/main/java/org/openmrs/util/Format.java]: If you want to break the convention, you will need to start a discussion on talk to defend your argument.
We even have a tool that generates a test for each should annotation. https://wiki.openmrs.org/display/docs/Generate+Test+Case+Plugin

Comment (winglam) [inline api/src/main/java/org/openmrs/util/Format.java]: I did not realize that each should annotation needs to correspond to a test. I have added/changed the should annotations so that each one corresponds to a test case now. Thanks.

**Comment by dkayiwa** [comment]
@winglam if you check again on the above wiki page, you will notice that the should annotation matches the method name.

**Comment by winglam** [comment]
Sorry, I seem to have overlooked that from before. I have now changed the method names to match the should annotation.

**Comment by dkayiwa** [comment]
@winglam hopefully you are aware of this plugin which should automatically generate test methods from the should annotations: https://wiki.openmrs.org/display/docs/Generate+Test+Case+Plugin
